mvcc/backend/backend.go

```diff
@@ -163,6 +165,20 @@ func (b *backend) ForceCommit() {
 }
 
 func (b *backend) Snapshot() Snapshot {
+	stopc, donec := make(chan struct{}), make(chan struct{})
+	go func() {
+		defer close(donec)
+		start := time.Now()
+		for {
+			select {
+			case <-time.After(snapshotWarningTimeout):
+				plog.Warningf("snapshotting is taking more than %v seconds to finish [started at %v]", time.Since(start).Seconds(), start)
```

Comment:
ah I missed that it was called in a loop.

* suggest use time.NewTicker instead of repeated time.After (less garbage) (don't forget to stop it when done)
* Still suggest printing the total time on completion ("snapshotting took 65.3s") but it's less important than I thought initially.